webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/ComposeService.scala

```diff
@@ -0,0 +1,169 @@
+package com.scalableminds.webknossos.datastore.services
+
+import com.scalableminds.util.geometry.Vec3Double
+import com.scalableminds.util.io.PathUtils
+import com.scalableminds.util.tools.{Fox, FoxImplicits}
+import com.scalableminds.webknossos.datastore.dataformats.n5.{N5DataLayer, N5SegmentationLayer}
+import com.scalableminds.webknossos.datastore.dataformats.precomputed.{
+  PrecomputedDataLayer,
+  PrecomputedSegmentationLayer
+}
+import com.scalableminds.webknossos.datastore.dataformats.wkw.{WKWDataLayer, WKWSegmentationLayer}
+import com.scalableminds.webknossos.datastore.dataformats.zarr.{ZarrDataLayer, ZarrSegmentationLayer}
+import com.scalableminds.webknossos.datastore.dataformats.zarr3.{Zarr3DataLayer, Zarr3SegmentationLayer}
+import com.scalableminds.webknossos.datastore.models.datasource.{
+  CoordinateTransformation,
+  DataLayer,
+  DataSource,
+  DataSourceId,
+  GenericDataSource
+}
+import net.liftweb.util.Helpers.tryo
+import play.api.libs.json.{Json, OFormat}
+
+import java.nio.charset.StandardCharsets
+import java.nio.file.{Files, Path}
+import javax.inject.Inject
+import scala.concurrent.ExecutionContext
+
+case class ComposeRequest(
+    newDatasetName: String,
+    targetFolderId: String,
+    dataStoreHost: String,
+    organizationName: String,
+    scale: Vec3Double,
+    layers: Seq[ComposeLayer]
+)
+
+object ComposeRequest {
+  implicit val composeRequestFormat: OFormat[ComposeRequest] = Json.format[ComposeRequest]
+}
+case class ComposeLayer(
+    id: DataSourceId,
+    sourceName: String,
+    newName: String,
+    transformations: Seq[CoordinateTransformation]
+)
+
+object ComposeLayer {
+  implicit val composeLayerFormat: OFormat[ComposeLayer] = Json.format[ComposeLayer]
+}
+
+class SymlinkHelper(dataSourceService: DataSourceService)(implicit ec: ExecutionContext) extends FoxImplicits {
```

Comment:
Here a trait with parameters would be nice (scala 3 feature). I could also remove this file and merge the content into UploadService, which would elimate this common trait/class.

Reply:
Could it be an injectable service? Then it could hold a proper dependency on the DataSourceService and be usable like the others?

Reply:
Move to another file?

Reply:
sounds good to me :)

Reply:
Did something a little more radical :wink: